#! /bin/sh /usr/share/dpatch/dpatch-run
## better_help_for_missing_module.dpatch by Daniel Hahler <ubuntu@thequod.de>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Better help in case the sysprof-module is not loaded. (LP: #30160)

@DPATCH@
diff -urNad sysprof-1.0.9~/sysprof.c sysprof-1.0.9/sysprof.c
--- sysprof-1.0.9~/sysprof.c	2007-10-21 23:43:43.000000000 +0200
+++ sysprof-1.0.9/sysprof.c	2008-01-23 23:21:09.027183686 +0100
@@ -560,7 +560,14 @@
 		       "\n"
 		       "       modprobe sysprof-module\n"
 		       "\n"
-		       "as root.");
+		       "as root.\n"
+		       "\n"
+		       "You may need to build and install this module first:\n"
+		       "  1. Install the package sysprof-module-source\n"
+		       "  2. Execute \"m-a a-i sysprof-module\"\n"
+		       "  3. Load the module as described above\n"
+		       "  4. Restart sysprof\n"
+		       "See also /usr/share/doc/sysprof-module-source/README.Debian" );
 		
 		update_sensitivity (app);
 		return;
